engine/src/configuration/game_config.h

```diff
@@ -40,49 +40,49 @@
 
 namespace pt = boost::property_tree;
 
-using std::string;
 using pt::ptree;
 
 // TODO: test this functionality, especially the subsection...
 
 class GameConfig {
 private:
-    static std::map<string, string> variables;
-    static string DEFAULT_ERROR_VALUE;
+    static std::string DEFAULT_ERROR_VALUE;
 
-    static inline string _GetVariable(string const &section, string const &name) {
-        string const key = section + "." + name;
-        if (variables.count(key)) {
-            return variables[key];
+    static inline std::string GetVar(std::string const &section, std::string const &name) {
+        std::string const key = section + "." + name;
```

Comment:
for consistency we should probably have a function that will do the concatenation for us, probably hand it a std::deque<std::string> as a parameter if we want to avoid varargs:

```C++
typedef qualifiedVarNameParams std::deque<std::string>
static inline std::string getQualifiedVarName(const qualifiedVarNameParams& orderedNameList) {
     std::string value = "";
     for (int i = 0; i < orderedNameList.size(); ++i) {
           if (!value.empty()) {
              value += ".";
           }
           value += orderedNameList[i];
     }
}
```
or something like that.

Doesn't have to be a std::deque; std::deque just offers random access and consistent ordering a result. std::list doesn't offer random access; and I'm not sure the iterators guarantee ordering, otherwise it could work too.

NOTE: Typically when I want to guarantee ordering I avoid iterators since you're beholden to how the iterator decides to traverse the data which can vary depending on the data-type. Random access via indexing always produces the same results, and the speed delta would only matter on super large (10,000+) record sets.

Even if this is only used internally for the name conversion it still helps with consistency and makes it easier to change out the separator.

(This could be a separator PR.)

Reply:
I am not against changing the type of parameter(s) we pass in for the section name(s) and setting name. However, I would very much like to leave this change to a future PR. Especially since I have several other branches queued up and waiting to turn into PRs at this point. ;-)

Reply:
In my later versions of the code, you pass the section; any subsection names; and the setting name all in a single, dot-separated string called `path`.

Another alternative might be to use `std::initializer_list` as the argument type. You would then call it something like this:

```C++11
GetVar({"graphics", "hud", "DiamondLineThickness"}, 1.0F);
```

Reply:
that can work too; I'm not tied to the specific data structure; that interface can certainly work too, let's just make sure it's well documented if we do.

Reply:
Agreed.